### ISSUE TITLE
Change how cardinalities are specified in the Metaschema

### ISSUE THREAD

**Issue by david-waltermire-nist** [usnistgov/OSCAL#441]
# User Story:

As an OSCAL model developer, I need to be able to define cardinalities in the Metaschema for ```assembly``` and ```field``` elements.

## Goals:

The current Metaschema allows for the cardinality of ```assembly``` and ```field``` elements to be defined using two capabilities. The first is by making the ```assembly``` and ```field``` plural and the second is by using the ```@required="yes"``` attribute. These two features result in the following:

|  | Singular | Plural |
| :--- | :--- | :--- |
| required="yes" | 1-1 | 1-many
| required="no" | 0-1 | 0-many

This approach differs from how cardinalities are defined in the XML and JSON schema languages.

To be more consistant the following changes need to be made:

- [ ] Minimum occurance needs to be specified using ```@min-occurs``` allowing a non-negative number value to be provided. This will replace the ```@required``` attribute. The default value for ```@min-occurs``` if not provided will be "0".

  For example: ```@min-occurs="0"``` or ```@min-occurs="1"```. 

- [ ] Maximum occurance needs to be specified using ```@max-occurs``` allowing a positive number value  or the constant value ```unbounded``` to be provided. This will allow for a set upper bound using a positive number, or any number of occurances using ```unbounded```. This will replace the use of ```assemblies``` and ```flags```. This approach adds the ability to set an actual upper bound, which is not currently supported in the Metaschema. The default value for ```@max-occurs``` if not provided will be "1".

  For example: ```@max-occurs="1"```, ```@max-occurs="5"```, or ```@max-occurs="unbounded"```.

- [ ] Generated XML and JSON schema need to be updated to support these new constructs.

  XML Schema uses the same constructs.
  
  JSON Schema uses similar constructs as follows:

  ```
  {
    "type": "array",
    "minItems": 2,
    "maxItems": 3
  }
  ```
  This must also work with the singlton/array optomization we are using in the generated schemas.
  
  - [ ] When ```@max-occurs``` is 1, the resulting schema must assume that only a singlton value will be provided. Otherwise, the schema will allow an array or a singleton.
  - [ ] When ```@min-occurs``` is 0 or 1, the resulting schema must allow for either a singleton or an array to be provided. If the ```@min-occurs``` is greater than 1, then the schema will allow only an array.

## Dependencies:

None.

## Acceptance Criteria

- [ ] All generated XML and JSON schema are updated to reflect these changes. Content instances are checked by the CI/CD workflow.
- [ ] XML and JSON documentation generated by the CI/CD workflow correctly represents the correct cardinalities.
- [ ] The Metaschema definitions are updated in a way that result in no changes to associated content.
- [ ] [Documentation](https://github.com/usnistgov/OSCAL/blob/master/src/metaschema/readme.md) on the use of Metaschema is updated to fully document use of the ```@min-occurs``` and ```max-occurs``` features.
- [ ] A single PR is approved that contains the above changes.

**Comment by iMichaela** [comment]
## 7/11/2019
This issue and the related ones (noted in #435 ) for the metaschema need to be reviewed together and unitested. However, more unit tests need to be created for the other features.

**Comment by david-waltermire-nist** [comment]
This will be closed once PR #464 is pulled.

**Comment by david-waltermire-nist** [comment]
This has been completed in PR #464.
